x-pack/plugins/uptime/public/components/monitor/synthetics/waterfall/components/use_flyout.ts

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useCallback, useState } from 'react';
+
+import {
+  ElementClickListener,
+  ProjectionClickListener,
+  ProjectedValues,
+  XYChartElementEvent,
+} from '@elastic/charts';
+
+import { WaterfallMetaData, WaterfallMetaDataEntry } from '../types';
+
+export const useFlyout = (metaData: WaterfallMetaData) => {
+  const [isFlyoutVisible, setIsFlyoutVisible] = useState(false);
+  const [flyoutData, setFlyoutData] = useState<WaterfallMetaDataEntry>({
```

Comment:
I mean to change this to undefined instead of a object that matches the shape with null values.